### PR TITLE
fix: made heartbeat client delete heartbeat after completing (MAPCO-5927)

### PR DESCRIPTION
**Pull request by nitzanmorr**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Further  information:

This fixes the bug that after completion the job would go back to Pending before being completed

### CODE REVIEW

File: MergerLogic/Clients/HeartbeatClient.cs [hunk 1]
```diff
@@ -44,13 +49,28 @@ public void Start(string taskId)
 
         public void Stop()
         {
-            if (!this._timer.Enabled)
+            try
             {
-                throw new Exception($"[{MethodBase.GetCurrentMethod().Name}] Heartbeat interval must be running in order to stop it.");
+                string relativeUri = $"heartbeat/remove";
```
Comment (CL-SHLOMIKONCHA) [inline MergerLogic/Clients/HeartbeatClient.cs:54]: Move the remove heartbeat after timer stopped

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: MergerLogic/Clients/HeartbeatClient.cs [hunk 2]
```diff
@@ -50,7 +55,22 @@ public void Stop()
...
+            {
+                string relativeUri = $"heartbeat/remove";
+                string url = new Uri(new Uri(this._baseUrl), relativeUri).ToString();
+                var content = JsonContent.Create(new[] { this._taskId });
+                this._httpClient.PostData(url, content);
+            }
+            catch (Exception e)
+            {
+                this._logger.LogError($"[{MethodBase.GetCurrentMethod().Name}] Could not delete heartbeat for task={this._taskId}, {e.Message}");
+                throw;
```
Comment (CL-SHLOMIKONCHA) [inline MergerLogic/Clients/HeartbeatClient.cs:68]: throw new Exception? throw e?

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: MergerLogic/Clients/HeartbeatClient.cs (hunk 2 above)
Comment (nitzanmorr) [reply MergerLogic/Clients/HeartbeatClient.cs]: fixed

File: MergerLogic/Clients/HeartbeatClient.cs (hunk 1 above)
Comment (nitzanmorr) [reply MergerLogic/Clients/HeartbeatClient.cs:60]: fixed

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Pull request merged**